### PR TITLE
cpu/esp: change dynamic SSID option handling

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR simplifies the definition of the option to use dynamic SSIDs for the SoftAP, especially with regard to the migration to `Kconfig`.

The semantics of defining a separate SSID prefix string that overrides the already defined SSID and enables the dynamic SSID generation with that prefix exactly when and only when it is set, made the handling of the parameter definition unnecessarily difficult and hard to understand.

Defining a boolean option that just enables the dynamic SSID generation, which then simply reuses the defined SSID as a prefix, makes it much more understandable and easier to handle, especially with respect to Kconfig.

This PR removes the configuration parameter `ESP_WIFI_PREFIX` and introduces the boolean configuration parameter `ESP_WIFI_SSID_DYNAMIC` which is defined as 0 by default. If set to 1, the already defined `ESP_WIFI_SSID` is used as prefix to generate a dynamic SSID by extending it with the MAC address of the SoftAP interface used, e.g.: `RIOT_AP_aabbccddeeff`

### Testing procedure

Compile and flash test application
```
USEMODULE=esp_wifi_ap \
CFLAGS='-DESP_WIFI_SSID=\"MySSID\" -DESP_WIFI_PASS=\"MyPassphrase\" -DESP_WIFI_MAX_CONN=1'  \
make -C examples/gnrc_networking BOARD=esp32-wroom-32 flash
```
with and without ` -DESP_WIFI_SSID_DYNAMIC=1` in `CFLAGS` and use any WiFi device to check the used SSID.

Without  ` -DESP_WIFI_SSID_DYNAMIC=1`, the SSID of the SoftAP should be `RIOT_AP`, with ` -DESP_WIFI_SSID_DYNAMIC=1`, the SSID of the SoftAP should be something like `RIOT_AP_<mac>`

### CODE REVIEW

**Review by jeandudey — APPROVED** [review]
ACK.

**Comment by gschorcht** [comment]
@jeandudey Can we merge it?

**Comment by jeandudey** [comment]
Yes, tested on `esp32-wroom-32` on both configs

File: cpu/esp_common/esp-wifi/esp_wifi_params.h [hunk 1]
```diff
@@ -44,26 +44,30 @@
...
- */
-#if !defined(ESP_WIFI_SSID) && !defined(ESP_WIFI_AP_PREFIX) || defined(DOXYGEN)
-#define ESP_WIFI_AP_PREFIX "RIOT_AP_"
-#endif
-
 /**
  * @brief   Passphrase used for the AP as clear text (max. 64 chars).
  */
-#ifdef DOXYGEN
+#ifndef ESP_WIFI_PASS
```
Comment (gschorcht) [inline cpu/esp_common/esp-wifi/esp_wifi_params.h:54]: `ESP_WIFI_PASS` was intentionally defined only when `DOXYGEN` is also defined to allow `ESP_WIFI_PASS` to be left undefined for using APs without authentication. There are no other changes in this PR affected by this change.

**Pull request merged**